presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolsExtractor.java

```diff
@@ -96,23 +95,6 @@ private SymbolsExtractor() {}
         return ImmutableSet.copyOf(extractAllVariable(expression, types));
     }
 
-    public static Set<VariableReferenceExpression> extractUniqueVariable(RowExpression expression, TypeProvider types)
```

Comment:
I think instead of removing this, you should remove the one in `GatherAndMergeWindows`. There will be more and more places that would need extract variable from `RowExpression`. It's better to handle them at one place. I'm actually changing this part actively (https://github.com/prestodb/presto/pull/12939/commits/aab2d6efa91930bddf6d0105a06e97221b741afe) so you can leave this commit out.

Reply:
That is already removed as well :). This method is no longer used anywhere. As discussed with @highker  earlier, we should either has everything in OriginalExpression or RowExpression. So the optimization rules call these function will be responsive for decide if the expression is RowExpression or not. This will make the check more explicit so any error throw is close to the source.